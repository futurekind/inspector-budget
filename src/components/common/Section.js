import React, { PropTypes } from 'react'
import styled from 'styled-components';
import Spacer from './Spacer'

const View = styled.section`
    text-align: ${props => props.textAlign};
`

const renderChildren = ({
    children,
    spacer
}) => {

    if(spacer) {
        return <Spacer {...spacer}>{ children }</Spacer>
    }

    return children
}

const Section = ({
    textAlign,
    children,
    spacer
}) => {

    return (
        <View textAlign={ textAlign }>
            { renderChildren({children, spacer}) }
        </View>
    )
}

Section.defaultProps = {
    textAlign: 'left'
}

Section.propTypes = {
    spacer: PropTypes.shape({
        type: PropTypes.oneOf([
            'topbottom', 'top', 'bottom'
        ]),
        value: PropTypes.oneOf([
            0, 1, 2, 3, 4, 5, 6, 7, 8
        ])
    }),
    textAlign: PropTypes.oneOf([
        'left', 'right', 'center'
    ]),
}

export default Section