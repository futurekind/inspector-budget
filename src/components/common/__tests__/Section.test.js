import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Section from '../Section';

describe('Section', () => {
    const wrapper = shallow(<Section />);
    const wrapperWithChildren = shallow(<Section><p>Foo</p></Section>);

    it('renders with default props', () => {
        expect(
            toJSON(wrapper)
        ).toMatchSnapshot()
    })

    it('renders textAlign prop', () => {
        wrapper.setProps({
            textAlign: 'center'
        })

        expect(toJSON(wrapper)).toMatchSnapshot()
    })

    it('renders its children', () => {
        expect(toJSON(wrapperWithChildren)).toMatchSnapshot()
    })

    it('renders its children with Spacer component', () => {
        wrapperWithChildren.setProps({
            spacer: {
                type: 'topbottom',
                value: 4
            }
        })
        expect(toJSON(wrapperWithChildren)).toMatchSnapshot()
    })
})