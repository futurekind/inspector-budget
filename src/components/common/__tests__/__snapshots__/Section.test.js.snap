exports[`Section renders its children 1`] = `
<styled.section
  textAlign="left">
  <p>
    Foo
  </p>
</styled.section>
`;

exports[`Section renders its children with Spacer component 1`] = `
<styled.section
  textAlign="left">
  <Spacer
    type="topbottom"
    value={4}>
    <p>
      Foo
    </p>
  </Spacer>
</styled.section>
`;

exports[`Section renders textAlign prop 1`] = `
<styled.section
  textAlign="center" />
`;

exports[`Section renders with default props 1`] = `
<styled.section
  textAlign="left" />
`;
